client/client.go

```diff
@@ -720,6 +720,9 @@ func (c *Client) reservePorts() {
 func (c *Client) fingerprint() error {
 	whitelist := c.config.ReadStringListToMap("fingerprint.whitelist")
 	whitelistEnabled := len(whitelist) > 0
+	blacklist := c.config.ReadStringListToMap("fingerprint.blacklist")
+	blacklistEnabled := len(blacklist) > 0
```

Comment:
White list needs this since you need to know both if the whitelist is active and if the fingerprinter is whitelisted (if not active nothing will be whitelisted). For blacklist you just need to know if it is contained in the map. so remove this variable and change :737 to `if _, ok := blacklist[name]; ok {`